endpoint/endpoint.go

```diff
@@ -369,23 +381,40 @@ func (e *Endpoint) CheckEndpoint() bool {
 
 func (t Targets) ValidateMXRecord() bool {
 	for _, target := range t {
-		// MX records must have a preference value to indicate priority, e.g. "10 example.com"
-		// as per https://www.rfc-editor.org/rfc/rfc974.txt
-		targetParts := strings.Fields(strings.TrimSpace(target))
-		if len(targetParts) != 2 {
-			log.Debugf("Invalid MX record target: %s. MX records must have a preference value to indicate priority, e.g. '10 example.com'", target)
-			return false
-		}
-		preferenceRaw := targetParts[0]
-		_, err := strconv.ParseUint(preferenceRaw, 10, 16)
+		_, err := ParseMXRecord(target)
 		if err != nil {
-			log.Debugf("Invalid SRV record target: %s. Invalid integer value in target.", target)
+			log.Debugf("Invalid MX record target: %s. %v", target, err)
 			return false
 		}
 	}
+
 	return true
 }
 
+func ParseMXRecord(target string) (MXTarget, error) {
+	var mxTarget MXTarget
+	// MX records must have a preference value to indicate priority, e.g. "10 example.com"
+	// as per https://www.rfc-editor.org/rfc/rfc974.txt
+	targetParts := strings.Fields(strings.TrimSpace(target))
+	if len(targetParts) != 2 {
+		err := fmt.Errorf("invalid MX record target: %s. MX records must have a preference value and a host, e.g. '10 example.com'", target)
```

Comment:
Can we have more idiomatic go, return null, error. Debug message we could have it where the method is used.